examples/inverse/plot_morph_stc_to_fsaverage_volume.py

```diff
@@ -0,0 +1,404 @@
+    # scenario, since those values were chosen to optimize computation time
+    stc.crop(0.087, 0.087)
+
+    # Save result in a 4D nifti file
+    img = mne.save_stc_as_volume(fname, stc, forward['src'],
+                                 mri_resolution=True)
+
+    return img
+
+
+###############################################################################
+# Compute non-linear morph mapping
+
+def compute_morph_map(img_m, img_s=None, niter_affine=(100, 100, 10),
+                      niter_sdr=(5, 5, 3)):
+    # get Static to world transform
+    img_s_grid2world = img_s.affine
+
+    # output Static as ndarray
+    img_s = img_s.dataobj[:, :, :]
+
+    # normalize values
+    img_s = img_s.astype('float') / img_s.max()
+
+    # get Moving to world transform
+    img_m_grid2world = img_m.affine
+
+    # output Moving as ndarray
+    img_m = img_m.dataobj[:, :, :]
+
+    # normalize values
+    img_m = img_m.astype('float') / img_m.max()
+
+    # compute center of mass
+    c_of_mass = imaffine.transform_centers_of_mass(img_s, img_s_grid2world,
+                                                   img_m, img_m_grid2world)
+
+    nbins = 32
+
+    # set up Affine Registration
+    affreg = imaffine.AffineRegistration(
+        metric=imaffine.MutualInformationMetric(nbins, None),
+        level_iters=list(niter_affine),
+        sigmas=[3.0, 1.0, 0.0],
+        factors=[4, 2, 1])
+
+    # translation
+    translation = affreg.optimize(img_s, img_m,
+                                  transforms.TranslationTransform3D(), None,
+                                  img_s_grid2world, img_m_grid2world,
+                                  starting_affine=c_of_mass.affine)
+
+    # rigid body transform (translation + rotation)
+    rigid = affreg.optimize(img_s, img_m,
+                            transforms.RigidTransform3D(), None,
+                            img_s_grid2world, img_m_grid2world,
+                            starting_affine=translation.affine)
+
+    # affine transform (translation + rotation + scaling)
+    affine = affreg.optimize(img_s, img_m,
+                             transforms.AffineTransform3D(), None,
+                             img_s_grid2world, img_m_grid2world,
+                             starting_affine=rigid.affine)
+
+    # apply affine transformation
+    img_m_affine = affine.transform(img_m)
+
+    # set up Symmetric Diffeomorphic Registration (metric, iterations)
+    sdr = imwarp.SymmetricDiffeomorphicRegistration(
+        metrics.CCMetric(3), list(niter_sdr))
+
+    # compute mapping
+    mapping = sdr.optimize(img_s, img_m_affine)
+
+    return mapping, affine
+
+
+###############################################################################
+# Save non linear mapping data
+
+def save_mapping(fname, data, overwrite=True):
+    out = dict()
+
+    # dissolve object structure
+    for d in data:
+        # save type for order independent decomposition
+        out[type(d).__name__] = d.__dict__
```

Comment:
Doing things with `__` attributes is generally not a good idea if it can otherwise be avoided.

Here you have two objects, you can explicitly just pass them (same with loading -- you know what should be in the dict returned by `load_mapping` file, just use them directly).